src/rpc.rs

```diff
@@ -60,14 +60,39 @@ pub struct JsonRpcResponse<T> {
 
 impl<T> JsonRpcResponse<T> {
     pub fn new(id: MessageId, result: T) -> Self {
-        JsonRpcResponse {
+        Self {
             id,
             jsonrpc: JSON_RPC_VERSION.clone(),
             result,
         }
     }
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct JsonRpcResponseError {
+    pub id: MessageId,
+    pub jsonrpc: Arc<str>,
+    pub error: JsonRpcError,
+}
+
+impl JsonRpcResponseError {
+    pub fn new(id: MessageId, error: JsonRpcError) -> Self {
+        Self {
+            id,
+            jsonrpc: JSON_RPC_VERSION.clone(),
+            error,
+        }
+    }
+}
+
+// https://www.jsonrpc.org/specification#error_object
+#[derive(Debug, Serialize, Deserialize)]
+pub struct JsonRpcError {
+    pub code: i16,
+    pub message: String,
+    // pub data: Value,
```

Comment:
why was this commented?

Reply:
JSON-RPC errors have a data field which can have more custom data. But we don't need it right now